hw2/04-background.js

```diff
@@ -1 +1,60 @@
-// Add your code here
+const bg = document.getElementById("background");
+const interval_element = document.getElementById("interval");
+const button = document.getElementById("start-stop");
+const message_element = document.getElementById("message");
+let interval_id;
+
+//The current button state
+// either a start button(true) or stop button(false)
+// should only be set by changeButtonState
+let is_button_start_state = true;
+
+//Toggle the button state, change it's color and text to match
+const changeButtonState = () => {
+  is_button_start_state = !is_button_start_state;
+  if (is_button_start_state) {
+    button.setAttribute("value", "Start");
+    button.setAttribute("class", "btn-primary");
+  } else {
+    button.setAttribute("value", "Stop");
+    button.setAttribute("class", "btn-danger");
+  }
+};
+
+//Generate a random color, with a reduced maximum alpha value to
+// prevent too harsh of colors
+const randomColor = () => {
+  const r = Math.round(Math.random() * 255);
+  const b = Math.round(Math.random() * 255);
+  const g = Math.round(Math.random() * 255);
+  const a = Math.round(Math.random() * 30) + 50;
+
```

Comment:
The function **randomColor** incorrectly generates the alpha channel of the color. The use of hexadecimal for the alpha is incorrect as CSS does not support alpha values in hexadecimal color codes.

It is mentioned in the requirements to use **rgba or hsla** instead for alpha transparency.

Reply:
It doesn't have to be right away, but can you clarify what you mean by this? On the next line the alpha value is the only one not being converted to hexadecimal as far as I can tell.

Reply:
The use of Math.round() will round the results to the nearest integer, potentially reaching 255 more often than using Math.floor() due to rounding up from any decimal .5 or higher.

For example,

>     const color = `rgba(${Math.floor(Math.random() * 256)}, ${Math.floor(
>       Math.random() * 256
>     )}, ${Math.floor(Math.random() * 256)}, 0.5)`;
>   };